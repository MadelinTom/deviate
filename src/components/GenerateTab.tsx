import { useState } from "react";

export const GenerateTab = ({onClick}: any) => {
  const [input, setInput] = useState(5000);

  return (
    <div
      id={"contentDiv"}
      style={{
        display: "flex",
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "center",
      }}
    >
      <div
        id={"form"}
        style={{
          background: "rgba(160, 158, 151, 0.8)",
          zIndex: 1,
          display: "flex",
          flexDirection: "row",
          borderRadius: "10px",
          alignItems: "center",
          margin: "50px",
        }}
      >
        <form style={{ zIndex: 2, margin: "10px", minWidth: "70px" }}>
          <label>Distance in km:</label>
          <br />
          <input
            type="number"
            id="distance"
            name="distance"
            onChange={(event) => setInput(parseInt(event.target.value))}
          />
        </form>

        <button
          style={{
            margin: "10px",
            minWidth: "150px",
            height: "50px",
            zIndex: 2,
          }}
          onClick={() => onClick(input)}
        >
          Generate
        </button>
      </div>
    </div>
  );
};
