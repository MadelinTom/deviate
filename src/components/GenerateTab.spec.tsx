import { render } from "@testing-library/react";
import { GenerateTab } from "./GenerateTab"

describe("The GenerateTab component", () => {
  it("should match the snapshot", () => {
    const component = render(<GenerateTab onClick={() => {}}/>);

    expect(component).toMatchSnapshot();
  })
})