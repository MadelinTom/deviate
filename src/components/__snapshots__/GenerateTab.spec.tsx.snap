// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The GenerateTab component should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="contentDiv"
        style="display: flex; flex-direction: row; align-items: center; justify-content: center;"
      >
        <div
          id="form"
          style="background: rgba(160, 158, 151, 0.8); z-index: 1; display: flex; flex-direction: row; border-radius: 10px; align-items: center; margin: 50px;"
        >
          <form
            style="z-index: 2; margin: 10px; min-width: 70px;"
          >
            <label>
              Distance in km:
            </label>
            <br />
            <input
              id="distance"
              name="distance"
              type="number"
            />
          </form>
          <button
            style="margin: 10px; min-width: 150px; height: 50px; border-radius: 10px; z-index: 2;"
          >
            Generate
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="contentDiv"
      style="display: flex; flex-direction: row; align-items: center; justify-content: center;"
    >
      <div
        id="form"
        style="background: rgba(160, 158, 151, 0.8); z-index: 1; display: flex; flex-direction: row; border-radius: 10px; align-items: center; margin: 50px;"
      >
        <form
          style="z-index: 2; margin: 10px; min-width: 70px;"
        >
          <label>
            Distance in km:
          </label>
          <br />
          <input
            id="distance"
            name="distance"
            type="number"
          />
        </form>
        <button
          style="margin: 10px; min-width: 150px; height: 50px; border-radius: 10px; z-index: 2;"
        >
          Generate
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
